scripts/scr_weapon/scr_weapon.gml

```diff
@@ -1821,59 +1821,157 @@ function equipment_struct(item_data, core_type,quality="none") constructor{
             self[$names[i]]=defaults[i];
         }
     }
-    spe_desc = special_description;
-    static special_description_gen = function(){
-        spe_desc+=" "
-        if (attack!=0){
-            spe_desc+=$"DAM {attack},#"
-        }
-        if (ranged_mod!=0){
-            spe_desc += $"Ranged {ranged_mod}%,#"
-        }
-        if (melee_mod!=0){
-            spe_desc += $"Melee {melee_mod}%,#"
-        }
-        if (hp_mod!=0){
-            spe_desc += $"HP {hp_mod}%,#"
-        }
-        if (damage_resistance_mod!=0){
-            spe_desc += $"Damage Res {damage_resistance_mod}%,#"
+
+    static item_tooltip_desc_gen = function(){
+        item_desc_tooltip = ""
+        var stat_order;
+        var item_type
+        if struct_exists(global.gear[$ "armour"],name){
+            item_type = "armour"
         }
-        if (armour_value!=0){
-            spe_desc += $"AC {armour_value},#"
+        else if struct_exists(global.gear[$ "mobility"],name){
```

Comment:
> the item type should already be defined in the weapon struct under "type" as such this check is not required just use the items stored value for what it is.

If I remember correctly, I've used this check instead of type because when the tooltip draw function is called in the shop screen - it doesn't have item type specified. I think shop screen and soldier screen use different sources of info or something.